packages/frontend/src/components/markdown-editor/markdown-editor.tsx

```diff
@@ -65,19 +67,51 @@ const emojiCompleter = {
 interface Props {
   contents: string;
   onContentsChange: (updatedValue: string) => any;
+  uploadFile?: (blob: File, name: string) => any;
 }
 
-export const MarkdownEditor = ({ contents, onContentsChange }: Props) => {
+export const MarkdownEditor = ({ contents, onContentsChange, uploadFile }: Props) => {
   const aceTheme = useColorModeValue('chrome', 'nord_dark');
 
   const [internalValue, setInternalValue] = useState(contents);
   const previousValueRef = useRef(contents);
 
+  const aceEditorRef: any = useRef();
+
   // Overwrite internal state if external contents change
   useEffect(() => {
     setInternalValue(contents);
   }, [contents]);
 
+  // Adding listener for the aceEditor on 'paste' event
+  useEffect(() => {
+    async function handlePasteEvent(arg: any) {
+      const items = (arg.clipboardData || arg.originalEvent.clipboardData).items;
+      if (items[0].type.startsWith('image/')) {
+        const blob = items[0].getAsFile();
+        if (uploadFile) {
+          const name = blob.name.startsWith('image.')
+            ? blob.name.replace('image', `pasted-${nanoid().substring(0, 6)}`)
+            : blob.name.replaceAll(' ', '-');
```

Comment:
There was also an issue when an image has a default name that contains spaces, that's why here we're replacing all spaces to dashes.